### PR TITLE
Bookmarks Persist After Toggle in Profile

### PR DESCRIPTION
**Pull request by sigmachirality**
this PR does the title and also removes localstorage usage since users can't see bookmarks unless they're logged in anyway.

### CODE REVIEW

File: components/ClubPage/Header.js [hunk 1]
```diff
@@ -27,7 +27,7 @@ const Header = ({ club, userInfo, favorites, style, updateFavorites }) => {
     (inClub && inClub.role <= ROLE_OFFICER) ||
     (userInfo && userInfo.is_superuser)
 
-  const [favCount, setFavCount] = useState(club.favorite_count || 0)
+  const [favCount] = useState(club.favorite_count || 0)
```
Comment (cacabo) [inline components/ClubPage/Header.js]: If not using the setter, why not just have
```suggestion
  const favCount = club.favorite_count || 0
```

File: components/Settings/FavoritesTab.js [hunk 2]
```diff
@@ -1,9 +1,17 @@
+import { useState } from 'react'
 import ClubTableRow from '../ClubTableRow'
 
-export default (props) => {
-  const { updateFavorites, favorites, clubs } = props
-
+export default ({ clubs, favorites, updateFavorites }) => {
+  const [table, setTable] = useState(() => {
+    const ret = {}
+    favorites.forEach(favorite => { ret[favorite] = true })
```
Comment (cacabo) [inline components/Settings/FavoritesTab.js]: I think prettier would recommend this but I might be wrong
```suggestion
    favorites.forEach(favorite => {
      ret[favorite] = true
    })
```

File: components/ClubPage/Header.js (hunk 1 above)
Comment (sigmachirality) [reply components/ClubPage/Header.js]: I think this lint warning is being handled in another PR

Comment (ezwang) [reply components/ClubPage/Header.js]: I believe that #61 might use the setter, we can handle this during the merge process.

File: components/Settings/FavoritesTab.js [hunk 3]
```diff
@@ -1,9 +1,19 @@
...
-
+export default ({ keyword, clubs, favorites, updateFavorites }) => {
+  const [table, setTable] = useState(() => {
+    const ret = {}
+    favorites.forEach(favorite => {
+      ret[favorite] = true
+    })
+    return ret
+  })
   const findClub = clubs ? code => clubs.find(club => club.code === code) : () => {}
```
Comment (cacabo) [inline components/Settings/FavoritesTab.js]: Feel like there's a better way to do this—even just splitting into lines and adding a comment that "if there is no code, there is no club to be found" would be a lot more readable than this ternary operator

**Comment by cacabo** [comment]
Other than that one comment this lgtm

File: components/Settings/FavoritesTab.js (hunk 3 above)
Comment (sigmachirality) [reply components/Settings/FavoritesTab.js]: rewritten in the commit "rewrite findClub to be clearer"

**Comment by sigmachirality** [comment]
@ezwang I fixed the bug, was using the list of favorites updated by the user instead of the list of favorites cached in state to check whether to display the placeholder text

**Pull request merged**